File: mobile/lib/models/transaction.dart
class Transaction {
  final String id;
  final String type;
  final double amount;
  final String accountId;
  final String accountName;
  final String? categoryId;
  final String categoryName;
  final String categoryIcon;
  final String categoryColor;
  final String transactionDate;
  final String remark;
  final String? toAccountId;

  Transaction({
    required this.id,
    required this.type,
    required this.amount,
    required this.accountId,
    required this.accountName,
    this.categoryId,
    required this.categoryName,
    required this.categoryIcon,
    required this.categoryColor,
    required this.transactionDate,
    required this.remark,
    this.toAccountId,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'] ?? '',
      type: json['type'] ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      accountId: json['account_id'] ?? '',
      accountName: json['account_name'] ?? '',
      categoryId: json['category_id'],
      categoryName: json['category_name'] ?? '',
      categoryIcon: json['category_icon'] ?? '',
      categoryColor: json['category_color'] ?? '#007AFF',
      transactionDate: json['transaction_date'] ?? '',
      remark: json['remark'] ?? '',
      toAccountId: json['to_account_id'],
    );
  }

  bool get isIncome => type == 'income';
  bool get isExpense => type == 'expense';
  bool get isTransfer => type == 'transfer';

  String get categoryEmoji {
    const map = {
      '餐饮': '🍽️',
      '交通': '🚗',
      '购物': '🛍️',
      '娱乐': '🎮',
      '居住': '🏠',
      '医疗': '💊',
      '教育': '📚',
      '工资': '💰',
      '奖金': '🎁',
      '理财': '📈',
    };
    return map[categoryName] ?? '💳';
  }
}

class TransactionListResponse {
  final List<Transaction> list;
  final int total;
  final int page;
  final int pageSize;

  TransactionListResponse({
    required this.list,
    required this.total,
    required this.page,
    required this.pageSize,
  });

  factory TransactionListResponse.fromJson(Map<String, dynamic> json) {
    return TransactionListResponse(
      list: (json['list'] as List<dynamic>?)
          ?.map((e) => Transaction.fromJson(e))
          .toList() ?? [],
      total: json['total'] ?? 0,
      page: json['page'] ?? 1,
      pageSize: json['page_size'] ?? 20,
    );
  }

  bool get hasMore => list.length >= pageSize;
}

File: web/src/api/transaction.ts
import { get, post, put, del } from '@/utils/request'

export interface Transaction {
  id: string
  type: 'income' | 'expense' | 'transfer'
  amount: number
  account_id: string
  category_id: string | null
  transaction_date: string
  remark: string
  images: string
  tags: string
  to_account_id: string | null
  member_id?: string | null
  paid_by_member_id?: string | null
  source: string
  reminder_id?: string | null
  lending_id?: string | null
  recurring_id: string | null
  // Nested objects from backend
  account?: {
    id: string
    name: string
    type: string
    icon: string
    color: string
  }
  category?: {
    id: string
    name: string
    type: string
    icon: string
    color: string
  }
  to_account?: {
    id: string
    name: string
  }
  // Computed getters for backwards compatibility
  account_name?: string
  category_name?: string
  category_icon?: string
  category_color?: string
}

export interface TransactionListResponse {
  list: Transaction[]
  total: number
  page: number
  page_size: number
}

export interface TransactionListParams {
  page?: number
  page_size?: number
  start_date?: string
  end_date?: string
  type?: string
  account_id?: string
  category_id?: string
  keyword?: string
}

export interface CreateTransactionParams {
  type: 'income' | 'expense' | 'transfer'
  amount: number
  account_id: string
  to_account_id?: string
  category_id?: string
  transaction_date: string
  remark?: string
  images?: string
  tags?: string
  member_id?: string
  paid_by_member_id?: string
}

export const transactionApi = {
  getList(params?: TransactionListParams): Promise<TransactionListResponse> {
    return get<TransactionListResponse>('/transactions', { params })
  },

  getById(id: string): Promise<Transaction> {
    return get<Transaction>(`/transactions/${id}`)
  },

  create(params: CreateTransactionParams): Promise<Transaction> {
    return post<Transaction>('/transactions', params)
  },

  update(id: string, params: CreateTransactionParams): Promise<Transaction> {
    return put<Transaction>(`/transactions/${id}`, params)
  },

  delete(id: string): Promise<void> {
    return del<void>(`/transactions/${id}`)
  },

  batchDelete(ids: string[]): Promise<void> {
    return post<void>('/transactions/batch-delete', { ids })
  }
}
